File: practice19/5.cpp
//https://www.nowcoder.com/practice/752409657fee4fddbac587fb30fd59b6?tpId=98&&tqId=32991&rp=8&ru=/activity/oj&qru=/ta/2019test/question-ranking
#include<iostream>
using namespace std;
int main()
{
	int i, a[10000], x = 0, num = 0;
	while (1)
	{
		cin >> a[x];
		x++;
		if (cin.get() == '\n')
		{
			break;
		}
	}
	for (i = 0; i < x - 1; i++)
	{
		if (a[i + 1] < a[i])
		{
			num++;
		}
	}
	if (num <= 1)
	{
		cout << "1" << endl;
	}
	else
	{
		cout << "0" << endl;
	}
	system("pause");
	return 0;
}

File: practice32/5.java


//https://www.nowcoder.com/practice/53e4c208b8cf497086ecd65ef45349bb?tpId=85&tqId=29835&tPage=1&rp=1&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        while (in.hasNext()) {
            long num = in.nextLong();
            long x = (long) Math.pow(num, 0.5);
            if (x * (x + 1) > num) {
                System.out.println(x - 1);
            } else {
                System.out.println(x);
            }
        }
    }
}